backend/coreapp/compiler_wrapper.py

```diff
@@ -56,6 +60,32 @@ class CompilerWrapper:
     def base_path():
         return settings.COMPILER_BASE_PATH
 
+    @staticmethod
+    def arch_from_compiler(compiler: str) -> Optional[str]:
+        cfg = compilers.get(compiler)
+        return cfg["arch"] if cfg else None
+
+    def cc_opts_from_command(compiler: str, compile_command: str) -> str:
+        cfg = compilers[compiler]
+        # TODO: use cfg for this?
```

Comment:
This is weird, ideally we'd grab every flag given to cc

Reply:
we must avoid grabbing at least `-o`, `$INPUT`/`$OUTPUT`, and the compiler command itself. Some of them don't make sense to grab, like `-G0`/`-non_shared` which are already part of the base compiler command, and `-I...`/` `-woff 123`, which are fairly harmless but bloats the compiler option UI. A list of allowed options, which we can get from the cfg's setting view, seemed like a reasonable option. Could also go the other way and try to filter out unwanted things, though, just a tad more work.

Reply:
Yeah, perhaps we could grab all `-W`, `-f`, `-g`, and `-O` flags?

Reply:
I'm fine with it as-is for now, but yeah I guess having it be part of the cfg is good too (though repeating stuff in each compiler cfg could get annoying)